hazelcast/src/main/java/com/hazelcast/map/impl/nearcache/NearCacheImpl.java

```diff
@@ -248,6 +241,41 @@ public boolean remove(Data key) {
     }
 
     @Override
+    public Object mapKeyToMarker(Data key) {
+        NearCacheRecord markerRecord = new NearCacheRecord(key, MARKER_VALUE);
+        NearCacheRecord previous = cache.put(key, markerRecord);
```

Comment:
shouldn't this be CAS too? what if some other thread managed to populate the near cache? then you will remove completely good value.